models/articles.model.js

```diff
@@ -1,16 +1,29 @@
 const db = require("../db/connection");
 const { checkTopicExists } = require("../db/seeds/utils");
 
-exports.fetchArticleById = async (article_id) => {
+exports.fetchArticleById = async (article_id, queries) => {
   const article = await db.query(
     `
  SELECT * FROM articles 
  WHERE article_id = $1
  `,
     [article_id]
   );
-
-  return { article: article.rows[0] };
+  if (queries.hasOwnProperty("comment_count")) {
+    const comments = await db.query(
+      `
+ SELECT article_id FROM comments 
+ `
+    );
+    let commentCount = 0;
+    comments.rows.forEach((comment) => {
+      if (comment.article_id === article.rows[0].article_id) {
+        commentCount += 1;
```

Comment:
Rather than doing this, we should be just querying the database and using JOINS in order to get our comment count. Take a look at the 'SQL Relationships' notes for some examples on how to do this. 🙂